src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -1,50 +1,149 @@
+import java.util.UUID;
 
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import codeu.model.data.Conversation;
+import codeu.model.data.Message;
+import codeu.model.data.User;
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.MessageStore;
+import codeu.model.store.basic.UserStore;
+
 public class AdminServlet extends HttpServlet {
 
-    private HashSet<String> admins;
+  private HashSet<String> admins;
+
+  private UserStore userStore;
+  private ConversationStore conversationStore;
+  private MessageStore messageStore;
+
+  @Override
+  public void init() throws ServletException {
+    super.init();
+
+    setAdminUsernames();
+
+    setUserStore(UserStore.getInstance());
+    setConversationStore(ConversationStore.getInstance());
+    setMessageStore(MessageStore.getInstance());
+  }
+
+  void setAdminUsernames() {
+    admins = new HashSet<String>();
+    admins.add("admin");
+    admins.add("daniel");
+    admins.add("leslie");
+    admins.add("serena");
+    admins.add("shana");
+    admins.add("kyra");
+  }
+
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
+  void setConversationStore(ConversationStore conversationStore) {
+    this.conversationStore = conversationStore;
+  }
+  void setMessageStore(MessageStore messageStore) {
+    this.messageStore = messageStore;
+  }
 
-    @Override
-    public void init() throws ServletException {
-        super.init();
+  void computeAdminStats(HttpServletRequest request) {
 
-        setAdminUsernames();
+    // Calculate information to be displayed
+
+    // Users
+    List<User> allUsers = userStore.getAllUsers();
+
+    String newestUser = "N/A";
+    String mostActiveUser = "N/A";
+
+    // get newest user
+    if (allUsers.size() > 0) {
+      newestUser = allUsers.get(allUsers.size() - 1).getName();
     }
 
-    void setAdminUsernames() {
-        admins = new HashSet<String>();
-        admins.add("daniel");
-        admins.add("leslie");
-        admins.add("serena");
-        admins.add("shana");
-        admins.add("kyra");
+
+    // Conversations
+    List<Conversation> allConversations = conversationStore.getAllConversations();
+    List<Message> messagesInConversation;
+
+    // HashMap to hold user IDs and the number of messages sent by that user
+    HashMap<UUID, Integer> userMessageCounts = new HashMap<UUID, Integer>();
+
+    int totalMessageCount = 0; // total number of messages sent by all users
+    int totalWordCount = 0; // total number of words sent by all users
+
+    for (Conversation c: allConversations) {
+
+      messagesInConversation = messageStore.getMessagesInConversation(c.getId());
+      totalMessageCount += messagesInConversation.size();
+
+      for (Message m: messagesInConversation) {
+
+        int tmp = userMessageCounts.getOrDefault(m.getAuthorId(), 0);
```

Comment:
`previousMessageCount` or `authorMessageCount` would be a more descriptive name than `tmp`.